src/app/components/playback/playback_control.rs

```diff
@@ -115,18 +115,43 @@ impl PlaybackControl {
         let model = Rc::new(model);
         let debouncer = Debouncer::new();
         let debouncer_clone = debouncer.clone();
-
         let track_position = &widget.track_position;
+
         widget.seek_bar.connect_change_value(
-            clone!(@weak model, @weak track_position => @default-return signal::Inhibit(false), move |_, _, requested| {
+            clone!(@weak model, @weak track_position => @default-return signal::Inhibit(false), move |_, scroll, mut requested| {
+                let mut debounce = 200;
+                match scroll {
+                    gtk::ScrollType::StepForward => {
+                        let duration = model.current_song_duration().unwrap_or(0.0);
+                        requested = if requested + 5000.0 <= duration {
```

Comment:
extract step value to a local variable or a constant

Reply:
or simply min(requested + step, duration)

Reply:
Yes I was also thinking to have step value be a setting in the future.

Reply:
as for the min max : have a look at https://doc.rust-lang.org/std/primitive.f64.html#method.max it should be much simpler :)